src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/CertificateRegistry.java

```diff
@@ -5,155 +5,81 @@
-        if (Utils.isEmpty(certificatePem)) {
-            throw new IllegalArgumentException("Certificate PEM is empty");
-        }
-        try {
-            Optional<String> certId = getAssociatedCertificateId(certificatePem).map(Optional::of)
-                    .orElseGet(() -> fetchActiveCertificateId(certificatePem));
-            updateRegistryForCertificate(certificatePem, certId);
-            return certId;
-        } catch (CloudServiceInteractionException e) {
-            return getAssociatedCertificateId(certificatePem).map(Optional::of)
-                    .orElseThrow(() -> e);
+    public Certificate getCertificateById(String certificateId) {
+        Instant lastValid = registry.get(certificateId);
+        if (lastValid != null) {
+            return new Certificate(certificateId, Certificate.Status.ACTIVE, lastValid);
         }
+        return new Certificate(certificateId, Certificate.Status.INACTIVE);
     }
 
     /**
-     * Retrieves Certificate ID from IoT Core.
-     *
-     * @param certificatePem Certificate PEM
-     * @return IoT Certificate ID or empty Optional if certificate is inactive or invalid
+     * Update certificate.
+     * @param certificate certificate object
      */
-    private Optional<String> fetchActiveCertificateId(String certificatePem) {
-        return iotAuthClient.getActiveCertificateId(certificatePem);
-    }
-
-    /**
-     * Returns IoT Certificate ID associated locally for given certificate PEM.
-     *
-     * @param certificatePem Certificate PEM
-     * @return Certificate ID or empty optional
-     */
-    private Optional<String> getAssociatedCertificateId(String certificatePem) {
-        return Optional.ofNullable(getCertificateHash(certificatePem))
-                .map(registry::get);
-    }
-
-    /**
-     * Locally caches IoT Certificate ID mapping for Certificate PEM.
-     *
-     * @param certificateId IoT Certificate ID
-     * @param certificatePem Certificate PEM
-     */
-    private void registerCertificateIdForPem(String certificateId, String certificatePem) {
-        Optional.ofNullable(getCertificateHash(certificatePem))
-                .ifPresent(certHash -> registry.put(certHash, certificateId));
-    }
-
-    /**
-     * Updates certPem <-> IoT Certificate ID association in the registry.
-     *
-     * @param certificatePem Certificate PEM
-     * @param iotCertificateId Optional of IoT Certificate ID
-     */
-    private void updateRegistryForCertificate(String certificatePem, Optional<String> iotCertificateId) {
-        if (iotCertificateId.isPresent()) {
-            registerCertificateIdForPem(iotCertificateId.get(), certificatePem);
+    public void updateCertificate(Certificate certificate) {
+        // NOTE: The storage aspect of this will change soon. But for now
+        //  only store ACTIVE certificates in the registry
+        if (certificate.getStatus() == Certificate.Status.ACTIVE) {
```

Comment:
Similar to my comment above this logic is making the repository, registry a bit smart and doing an unexpected operation. When I call update on something I wouldn't expect it to get removed from storage

Reply:
Ack. Can add a remove API